### ISSUE TITLE
Cannot disable SARIF report with detekt.yaml

### ISSUE THREAD

**Issue by wujek-srujek** [detekt/detekt#6464]
## Expected Behavior

Following the documentation: https://detekt.dev/docs/introduction/configurations#output-reports if I exclude all reports, SARIF is still created (excluding all the others works fine so this is not an issue of 'Detekt is not finding my config file'). There seems to be no way to exclude SARIF in detekt.yaml.

## Observed Behavior

A SARIF report is generated even though it is excluded.

## Your Environment

* Version of detekt used: 1.23.1
* Version of Gradle used (if applicable): 8.3
* Operating System and version: macOS Ventura 13.5.1 on Apple M1 Pro

**Comment by 3flex** [comment]
We'll need more information, as this is unfortunately not enough to go on. A full reproducer is most helpful.

**Comment by wujek-srujek** [comment]
This is trivial to reproduce. Steps:

1. Create a new project. In my case, I created a new empty Android application with Android Studio Giraffe | 2022.3.1 Patch 1. Build configuration language: Groovy DSL (but I believe this makes no difference).
1. Integrate Detekt 1.23.1 to a project:
   - Create the following file in the `app` folder:
     ```yaml
     output-reports:
     active: true
     exclude:
       - 'HtmlOutputReport'
       - 'MdOutputReport'
       - 'SarifOutputReport'
       - 'TxtOutputReport'
       - 'XmlOutputReport'
     ```
   - Register Detekt Gradle plugin in `app/build.gradle` (somewhere at the top):
     ```groovy
     plugins {
       // Other plugins
       id 'io.gitlab.arturbosch.detekt' version '1.23.1'
     }
     ```
   - Add to `app/build.gradle`:
     ```groovy
     detekt {
       basePath = projectDir
       config.setFrom(file('detekt.yaml'))
       buildUponDefaultConfig = true
       ignoreFailures = true
     }
     ```
1. Run ` ./gradlew clean detekt` in the main folder of the project created above.
1. Run `ls app/build/reports/detekt/`.

Actual behavior: even though I disabled all reports, the `ls` from above still shows the SARIF report has been created.

Expected behavior: SARIF report is not created.

**Comment by atulgpt** [comment]
@wujek-srujek as a workaround setting below in your gradle file will stop the sarif file
```gradle
tasks {
    withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
        jvmTarget = "1.8"
        reports {
            html.required.set(false)
            txt.required.set(false)
            sarif.required.set(false)
            xml.required.set(false)
            md.required.set(false)
        }
    }
}
```

I am able to repro your issue and only the above code is fixing the issue for me

**Comment by lexa-diky** [comment]
SarifOutputReport has its id set to just `sarif` while all other report have their name 'inherited' from class name. You can user following codec to disable sarif reports in config:

```yaml
output-reports:
active: true
exclude:
  - 'HtmlOutputReport'
  - 'MdOutputReport'
  - 'sarif'
  - 'TxtOutputReport'
  - 'XmlOutputReport'
```

**Comment by lexa-diky** [comment]
[Suggested fix](https://github.com/detekt/detekt/pull/6479) - very breaking change

**Comment by atulgpt** [comment]
We should take this given `detekt` 2.0 we are taking these types of changes cc @cortinico @BraisGabin

**Comment by lexa-diky** [comment]
Created 2 semi-related issues 

#6481 - issue for removal of default value for `Extension::id`
#6482 - issue for removal/rename of `OutputReport::name`

**Comment by lexa-diky** [comment]
https://github.com/detekt/detekt/pull/6484 - documentation and default config fix to for 1.x.

**Comment by BraisGabin** [comment]
> I also would suggest:
> - removing default implementation from Extension::id to very implicit nature of it's behavior
> - removing OutputReport::name, because it is practically useless and makes implementing report less straightforward + current documentation on it is incorrect 

I agree with both suggestions. Do you want to send PRs for that?

**Comment by atulgpt** [comment]
@lexa-diky should this be closed since your PRs are merged?

**Comment by cortinico** [comment]
> @lexa-diky should this be closed since your PRs are merged?

https://github.com/detekt/detekt/pull/6484 is not merged

**Comment by lexa-diky** [comment]
closed #6484 and opened #6579 into main branch. Now docs reflect **current** behavior correctly

@cortinico should I rename "sarif" report SarifOutputReport? So it matches naming scheme of other reports?

**Comment by cortinico** [comment]
I'd say yes, once the docs are updated, let's rename the `SarifOutputReport` on `main` so for 2.x we'll have it fixed 👍

**Comment by BraisGabin** [comment]
Fixed by #6579
